addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -304,9 +323,22 @@ function factory(dependencies) {
                 fetchedMessages: messages,
                 threadCache: this,
             });
+            for (const threadView of this.threadViews) {
+                threadView.addComponentHint('messages-loaded', { fetchedMessages: messages });
+            }
```

Comment:
It would be best if the hint stayed where it was, the duplicate hint with more loaded and new loaded is not a good idea.

Reply:
I think you can revert now if you throw the error in catch instead of returning false.